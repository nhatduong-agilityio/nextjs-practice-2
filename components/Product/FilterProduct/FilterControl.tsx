import { Button, Heading, Stack } from '@chakra-ui/react'
import React, { memo } from 'react'

interface IProps {
  onHandleOpenFilter: () => void
}

const FilterControl = memo(({ onHandleOpenFilter }: IProps) => (
  <Stack flexDir='row' justifyContent='space-between' alignItems='center' mb={['20px', '50px']}>
    <Heading variant={['extraBoldMD', 'extraBoldXL']}>Product</Heading>
    <Button
      display={['flex', 'none']}
      color='black'
      variant='ghost'
      fontSize='14px'
      width='max-content'
      height='fit-content'
      style={{
        paddingInline: 0,
      }}
      onClick={onHandleOpenFilter}
      justifyContent='flex-end'
    >
      Filter
    </Button>
  </Stack>
))

FilterControl.displayName = 'FilterControl'

export default FilterControl
