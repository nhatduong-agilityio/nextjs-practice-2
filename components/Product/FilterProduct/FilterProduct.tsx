import { v4 as uuidv4 } from 'uuid'

import { FormControl, FormLabel, Stack } from '@chakra-ui/react'
import { useSearchParams } from 'next/navigation'
import React, { memo, useMemo } from 'react'
import { CategoriesMock, ColorMock } from '@mock/dataMock'
import type { TProducts } from '@matched-types/product'
import Dropdown from './Dropdown'

export const ratingOptions = [
  { id: uuidv4(), value: 1, label: '1 Star Up' },
  { id: uuidv4(), value: 2, label: '2 Star Up' },
  { id: uuidv4(), value: 3, label: '3 Star Up' },
  { id: uuidv4(), value: 4, label: '4 Star Up' },
  { id: uuidv4(), value: 5, label: '5 Star' },
]

type TOption = {
  id: string
  label: string
  value: string | number
}

const convertCategoryOption = (obj: Object): TOption[] =>
  Object.entries(obj).map(([item, itemValue]) => ({
    id: uuidv4(),
    label: item,
    value: itemValue.NAME,
  }))

const categoryOptions = Object.entries(CategoriesMock).map(([key]) => {
  const chairOption = convertCategoryOption(CategoriesMock.CHAIR)
  const destOption = convertCategoryOption(CategoriesMock.DESK)
  const bedOption = convertCategoryOption(CategoriesMock.Bed)
  const carpetOption = convertCategoryOption(CategoriesMock.CARPET)
  const sofaOption = convertCategoryOption(CategoriesMock.SOFA)
  const cupboardOption = convertCategoryOption(CategoriesMock.CUPBOARD)

  return {
    id: uuidv4(),
    label: key,
    value: [chairOption, destOption, bedOption, carpetOption, sofaOption, cupboardOption],
  }
})

export const priceOptions = [
  { id: uuidv4(), value: 1, label: 'Maximum Price' },
  { id: uuidv4(), value: 2, label: 'Minimum Price' },
]

interface IFilterProduct {
  products: TProducts
}

const FilterProduct = memo(({ products }: IFilterProduct): JSX.Element => {
  const searchParams = useSearchParams()!

  const search = useMemo(
    () => searchParams && searchParams.get('category')?.toString(),
    [searchParams],
  )

  const productCategories = useMemo(() => products?.map((product) => product.category), [products])

  return (
    <Stack style={{ margin: 0 }} w='full'>
      <FormControl w='full'>
        <FormLabel fontSize={['20px', '20px']} fontWeight='bold'>
          Filter
        </FormLabel>

        <Dropdown
          name='Color'
          items={ColorMock as TOption[]}
          type='checkBox'
          search={search}
          selectedCategory={productCategories}
        />
        <Dropdown
          name='Rating'
          items={ratingOptions}
          type='checkBox'
          search={search}
          selectedCategory={productCategories}
        />
        <Dropdown
          name='Price'
          items={priceOptions}
          type='input'
          search={search}
          selectedCategory={productCategories}
        />
        <Dropdown
          name='Category'
          nestedItems={categoryOptions}
          type='button'
          search={search}
          selectedCategory={productCategories}
        />
      </FormControl>
    </Stack>
  )
})

FilterProduct.displayName = 'FilterProduct'

export default FilterProduct
