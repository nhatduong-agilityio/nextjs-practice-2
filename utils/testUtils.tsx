import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
