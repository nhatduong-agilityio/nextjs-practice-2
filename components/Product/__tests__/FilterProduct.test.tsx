// Utils
import { render } from '@utils/testUtils'

import { ProductsMock } from '@mock/dataMock'

// Components
import FilterProduct from '../FilterProduct/FilterProduct'

jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(),
}))

describe('FilterProduct render', () => {
  const propsMock = {
    products: ProductsMock,
  }

  it('Should FilterProduct match Snapshot', () => {
    const { container } = render(<FilterProduct {...propsMock} />)

    expect(container).toMatchSnapshot()
  })
})
