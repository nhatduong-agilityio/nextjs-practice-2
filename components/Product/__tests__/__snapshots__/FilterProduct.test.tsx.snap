// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterProduct render Should FilterProduct match Snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-ppmla1"
    style="margin: 0px;"
  >
    <div
      class="chakra-form-control css-r4izcz"
      role="group"
    >
      <label
        class="chakra-form__label css-tdupph"
        for="field-:r0:"
        id="field-:r0:-label"
      >
        Filter
      </label>
      <div
        class="chakra-stack css-1v4uoso"
        style="margin: 0px;"
      >
        <button
          class="chakra-button css-l43bjg"
          style="padding-inline: 0;"
          type="button"
        >
          color
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-pfz1im"
              focusable="false"
              viewBox="0 0 8 13"
            >
              <path
                d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
              />
            </svg>
          </span>
        </button>
        <div
          class="chakra-stack css-19pl6ik"
          style="margin: 0px;"
        >
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="Blue"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="blue"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              Blue
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="Purple"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="purple"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              Purple
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="Red"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="red"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              Red
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="Orange"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="orange"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              Orange
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="Yellow"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="yellow"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              Yellow
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="Green"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="green"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              Green
            </p>
          </label>
        </div>
      </div>
      <div
        class="chakra-stack css-1v4uoso"
        style="margin: 0px;"
      >
        <button
          class="chakra-button css-l43bjg"
          style="padding-inline: 0;"
          type="button"
        >
          rating
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-pfz1im"
              focusable="false"
              viewBox="0 0 8 13"
            >
              <path
                d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
              />
            </svg>
          </span>
        </button>
        <div
          class="chakra-stack css-19pl6ik"
          style="margin: 0px;"
        >
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="1 Star Up"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="1"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              1 Star Up
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="2 Star Up"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="2"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              2 Star Up
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="3 Star Up"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="3"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              3 Star Up
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="4 Star Up"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="4"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              4 Star Up
            </p>
          </label>
          <label
            class="css-d3wqq2"
          >
            <input
              aria-disabled="false"
              aria-invalid="false"
              hidden=""
              id="field-:r0:"
              name="5 Star"
              style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
              type="checkbox"
              value="5"
            />
            <div
              aria-hidden="true"
              class="css-i2vche"
            />
            <p
              class="chakra-text css-5l0nl"
            >
              5 Star
            </p>
          </label>
        </div>
      </div>
      <div
        class="chakra-stack css-1v4uoso"
        style="margin: 0px;"
      >
        <button
          class="chakra-button css-l43bjg"
          style="padding-inline: 0;"
          type="button"
        >
          price
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-pfz1im"
              focusable="false"
              viewBox="0 0 8 13"
            >
              <path
                d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
              />
            </svg>
          </span>
        </button>
        <div
          class="chakra-stack css-19pl6ik"
          style="margin: 0px;"
        >
          <form
            style="margin-top: 0px; width: 100%;"
          >
            <div
              class="chakra-form-control css-1wvres7"
              role="group"
              style="margin: 0px;"
            >
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <div
                  class="chakra-input__left-element css-1vbpnog"
                >
                  <span
                    class="chakra-text css-kcidfd"
                  >
                    Price
                  </span>
                </div>
                <input
                  class="chakra-input css-ueq02j"
                  id="field-:r1:"
                  placeholder="Maximum Price"
                  type="number"
                />
              </div>
            </div>
            <div
              class="chakra-form-control css-1wvres7"
              role="group"
              style="margin: 0px;"
            >
              <div
                class="chakra-input__group css-bx0blc"
                data-group="true"
              >
                <div
                  class="chakra-input__left-element css-1vbpnog"
                >
                  <span
                    class="chakra-text css-kcidfd"
                  >
                    Price
                  </span>
                </div>
                <input
                  class="chakra-input css-ueq02j"
                  id="field-:r2:"
                  placeholder="Minimum Price"
                  type="number"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="chakra-stack css-1v4uoso"
        style="margin: 0px;"
      >
        <button
          class="chakra-button css-l43bjg"
          style="padding-inline: 0;"
          type="button"
        >
          category
          <span
            class="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-pfz1im"
              focusable="false"
              viewBox="0 0 8 13"
            >
              <path
                d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
              />
            </svg>
          </span>
        </button>
        <div
          class="chakra-stack css-19pl6ik"
          style="margin: 0px;"
        >
          <div
            class="chakra-stack css-1v4uoso"
            style="margin: 0px;"
          >
            <button
              class="chakra-button css-lbmdfj"
              style="padding-inline: 0;"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-pfz1im"
                  focusable="false"
                  style="margin-right: 13px;"
                  viewBox="0 0 8 13"
                >
                  <path
                    d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
                  />
                </svg>
              </span>
              chair
            </button>
            <div
              class="chakra-stack css-19pl6ik"
              style="margin: 0px;"
            >
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Chair"
                >
                  <button
                    class="chakra-button css-1nzk6tf"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Chair
                  </button>
                </a>
              </div>
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Office+Chair"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Office Chair
                  </button>
                </a>
              </div>
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Family+Chair"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Family Chair
                  </button>
                </a>
              </div>
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Reading+Chair"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Reading Chair
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1v4uoso"
            style="margin: 0px;"
          >
            <button
              class="chakra-button css-lbmdfj"
              style="padding-inline: 0;"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-pfz1im"
                  focusable="false"
                  style="margin-right: 13px;"
                  viewBox="0 0 8 13"
                >
                  <path
                    d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
                  />
                </svg>
              </span>
              dest
            </button>
            <div
              class="chakra-stack css-19pl6ik"
              style="margin: 0px;"
            >
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Dest"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Dest
                  </button>
                </a>
              </div>
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Office+dest"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Office dest
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1v4uoso"
            style="margin: 0px;"
          >
            <button
              class="chakra-button css-lbmdfj"
              style="padding-inline: 0;"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-pfz1im"
                  focusable="false"
                  style="margin-right: 13px;"
                  viewBox="0 0 8 13"
                >
                  <path
                    d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
                  />
                </svg>
              </span>
              best
            </button>
            <div
              class="chakra-stack css-19pl6ik"
              style="margin: 0px;"
            >
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Best"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Best
                  </button>
                </a>
              </div>
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Springbed"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Springbed
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1v4uoso"
            style="margin: 0px;"
          >
            <button
              class="chakra-button css-lbmdfj"
              style="padding-inline: 0;"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-pfz1im"
                  focusable="false"
                  style="margin-right: 13px;"
                  viewBox="0 0 8 13"
                >
                  <path
                    d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
                  />
                </svg>
              </span>
              carpet
            </button>
            <div
              class="chakra-stack css-19pl6ik"
              style="margin: 0px;"
            >
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Carpet"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Carpet
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1v4uoso"
            style="margin: 0px;"
          >
            <button
              class="chakra-button css-lbmdfj"
              style="padding-inline: 0;"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-pfz1im"
                  focusable="false"
                  style="margin-right: 13px;"
                  viewBox="0 0 8 13"
                >
                  <path
                    d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
                  />
                </svg>
              </span>
              sofa
            </button>
            <div
              class="chakra-stack css-19pl6ik"
              style="margin: 0px;"
            >
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Sofa"
                >
                  <button
                    class="chakra-button css-1nzk6tf"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Sofa
                  </button>
                </a>
              </div>
            </div>
          </div>
          <div
            class="chakra-stack css-1v4uoso"
            style="margin: 0px;"
          >
            <button
              class="chakra-button css-lbmdfj"
              style="padding-inline: 0;"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-icon css-pfz1im"
                  focusable="false"
                  style="margin-right: 13px;"
                  viewBox="0 0 8 13"
                >
                  <path
                    d="M0.612558 0.868804C0.68513 0.796049 0.771341 0.738325 0.866256 0.69894C0.96117 0.659555 1.06292 0.639282 1.16568 0.639282C1.26845 0.639282 1.3702 0.659555 1.46511 0.69894C1.56003 0.738325 1.64624 0.796049 1.71881 0.868804L6.91256 6.06255C6.9705 6.12038 7.01647 6.18906 7.04783 6.26466C7.07919 6.34027 7.09534 6.42132 7.09534 6.50318C7.09534 6.58503 7.07919 6.66609 7.04783 6.74169C7.01647 6.8173 6.9705 6.88598 6.91256 6.9438L1.71881 12.1376C1.41256 12.4438 0.918808 12.4438 0.612558 12.1376C0.306308 11.8313 0.306308 11.3376 0.612558 11.0313L5.13756 6.50005L0.606309 1.9688C0.306309 1.6688 0.306308 1.1688 0.612558 0.868804Z"
                  />
                </svg>
              </span>
              cupboard
            </button>
            <div
              class="chakra-stack css-19pl6ik"
              style="margin: 0px;"
            >
              <div
                class="css-ie9luo"
              >
                <a
                  href="/product?category=Cupboard"
                >
                  <button
                    class="chakra-button css-1kezzwz"
                    style="padding-inline: 0;"
                    type="button"
                  >
                    Cupboard
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
